README.md

```diff
@@ -28,12 +30,12 @@ To start learning about Kokkos:
 
 The latest release of Kokkos can be obtained from the [GitHub releases page](https://github.com/kokkos/kokkos/releases/latest).
 
-The current release is [4.2.01](https://github.com/kokkos/kokkos/releases/tag/4.2.01).
+The current release is [4.3.00](https://github.com/kokkos/kokkos/releases/tag/4.3.00).
```

Comment:
I didn't realize the recent releases were tracked here; I updated the release process doc with a reminder to update these during release-candidate branch creation